src/main/java/com/spartronics4915/lib/lidar/LidarProcessor.java

```diff
@@ -259,25 +259,29 @@ private void processLidarScan(LidarScan scan)
     {
         try
         {
-            Pose2d pose;
-            Twist2d vel;
-            if(mMode == OperatingMode.kRelative)
+            final RobotStateMap.State lastState = mLidarStateMap.getLatestState();
```

Comment:
@dbadb I think we should probably use `mEncoderStateMap` here... Because we feed our estimates back into `mLidarStateMap`, it's easy for use to get "carried away", and enter a positive feedback loop where we feed in a bad pose, which results in a bad guess for a pose, which results in an even worse pose, etc.

Also, why are we doing `getLatestState`? Shouldn't we be doing `mEncoderStateMap.get(scan.getTimestamp())`? Are you concerned the interpolation will result in bad guesses?